src/dynarec/rv64/dynarec_rv64_helper.h

```diff
@@ -104,14 +127,19 @@
             ORI(xFlags, xFlags, 1 << F_AF);                               \
         }                                                                 \
         IFX(X_CF) {                                                       \
-            /* cf = bc & (rex.w?(1<<63):(1<<31)) */                       \
-            SRLI(scratch1, scratch2, rex.w?63:31);                        \
+            /* cf = bc & (1<<(width-1)) */                                \
+            if (width == 8) {                                             \
+                ANDI(scratch1, scratch2, 0x80);                           \
+            } else {                                                      \
+                SRLI(scratch1, scratch2, width-1);                        \
+                if (width == 16) ANDI(scratch1, scratch1, 1);             \
```

Comment:
What is this special case with width==16?

Reply:
Oh, it's because for witdh==32 or 64, the ANDI is useless? Is that really true for width==32?

Reply:
I suppose the ZEROUP has already be called?

Reply:
> I suppose the ZEROUP has already be called?

Yes, when width==32, ZEROUP is called outside.

Reply:
yeah, 16bits operation are not very common on 64bits programs...